debezium/src/main/scala/akka/stream/alpakka/debezium/scaladsl/DebeziumSource.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.debezium.scaladsl
+
+import java.util
+import java.util.concurrent.{ExecutorService, Executors}
+
+import akka.NotUsed
+import akka.actor.{ActorSystem, Props}
+import akka.dispatch.ExecutionContexts
+import akka.stream.alpakka.debezium.impl.CommitterActor
+import akka.stream.alpakka.debezium.{CommittableRecord, DebeziumSettings, SimpleRecord}
+import akka.stream.scaladsl._
+import akka.stream.{OverflowStrategy, QueueOfferResult}
+import io.debezium.embedded.Connect
+import io.debezium.engine.DebeziumEngine
+import org.apache.kafka.connect.source.SourceRecord
+
+import scala.collection.JavaConverters._
+
+object DebeziumSource {
+  def committable(settings: DebeziumSettings, executor: ExecutorService = Executors.newSingleThreadExecutor())(
+      implicit s: ActorSystem
+  ): Source[CommittableRecord, NotUsed] = {
+    val (queue, source) =
+      Source
+        .queue[List[CommittableRecord]](settings.maxInMemoryElements, OverflowStrategy.backpressure)
+        .mapConcat(identity)
+        .preMaterialize()
+
+    val configurations = settings.properties
+
+    val valueConverter = settings.internalValueConverter.getDeclaredConstructor().newInstance()
+    valueConverter.configure(configurations.asInstanceOf[java.util.Map[String, String]], false)
+
+    val engine = DebeziumEngine
+      .create[SourceRecord](classOf[Connect])
+      .using(configurations)
+      .notifying((records: util.List[SourceRecord], committer: DebeziumEngine.RecordCommitter[SourceRecord]) => {
+        val list = records.asScala.toList
+        val committerActor = s.actorOf(Props(new CommitterActor(list, committer)))
+
+        queue
+          .offer(list.map(r => new CommittableRecord(r, committerActor, valueConverter)))
+          .map {
+            case QueueOfferResult.Enqueued =>
+            //actually do nothing
+            case _ =>
+              throw new InterruptedException("error while enqueueing records")
+          }(ExecutionContexts.sameThreadExecutionContext)
+      })
+      .using(
+        (success: Boolean, message: String, error: Throwable) =>
+          if (!success) queue.fail(error)
+          else queue.complete()
+      )
+      .build();
+
+    executor.execute(engine)
```

Comment:
Source as a custom graph stage probably would be better, but, I'm not quite sure how I'd fit since it's Debezium that pushes data into the Source, and not the Source that tries to pull it. That's why I thought a queue it'd fit a little better.
I guess I'll try to hack Debezium by creating a new custom engine that does not run by itself but rather expose a method for us to pulls data. I'm just not quite sure if this is straightforward tho.